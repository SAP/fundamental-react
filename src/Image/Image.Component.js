import { Image } from '../';
import React from 'react';
import { Description, DocsText, DocsTile, Header, Import, Playground, Properties, Separator } from '../_playground';

export const ImageComponent = () => {
    const sizesImageCode = `<Image size="s" photo="https://placeimg.com/400/400/nature"></Image>
<Image size="m" photo="https://placeimg.com/400/400/nature"></Image>
<Image size="l" photo="https://placeimg.com/400/400/nature"></Image>`;

    const shapesImageCode = `<Image size="s" type="circle" photo="https://placeimg.com/400/400/nature"></Image>
<Image size="m" type="circle" photo="https://placeimg.com/400/400/nature"></Image>
<Image size="l" type="circle" photo="https://placeimg.com/400/400/nature"></Image>`;
    return (
        <div>
            <Header>Image</Header>
            <Description>
                When using images, use the following helpers classes to adjust the size and the shape.
            </Description>
            <Import module='Image' path='/fundamental-react/src/' />

            <Separator />

            <Properties
                properties={[
                    {
                        name: 'size',
                        description:
                            'string (required)- the size of the image. Size options include \'s\' (24x24), \'m\' (36x36), and \'l\' (48x48).'
                    },
                    { name: 'photo', description: 'string (required) - picture url.' },
                    { name: 'type', description: 'string - When set to \'circle\' renders a round image.' }
                ]}
                type='Inputs' />

            <Separator />

            <h2>Sizes</h2>
            <DocsTile centered>
                <div className='fd-doc__margin--image'>
                    <Image photo='https://placeimg.com/400/400/nature' size='s' />
                    <Image photo='https://placeimg.com/400/400/nature' size='m' />
                    <Image photo='https://placeimg.com/400/400/nature' size='l' />
                </div>
            </DocsTile>
            <DocsText>{sizesImageCode}</DocsText>

            <Separator />

            <h2>Shapes</h2>
            <DocsTile centered>
                <div className='fd-doc__margin--image'>
                    <Image photo='https://placeimg.com/400/400/nature' size='s'
                        type='circle' />
                    <Image photo='https://placeimg.com/400/400/nature' size='m'
                        type='circle' />
                    <Image photo='https://placeimg.com/400/400/nature' size='l'
                        type='circle' />
                </div>
            </DocsTile>
            <DocsText>{shapesImageCode}</DocsText>

            <Separator />
            <h2>Playground</h2>
            <Playground
                component='image'
                schema={[
                    {
                        attribute: 'size',
                        typeOfAttribute: 'string',
                        'enum': ['s', 'm', 'l']
                    },
                    {
                        attribute: 'photo',
                        typeOfAttribute: 'string'
                    },
                    {
                        attribute: 'type',
                        typeOfAttribute: 'string',
                        'enum': ['', 'circle']
                    }
                ]}>
                <Image photo='https://placeimg.com/400/400/nature' size='s'
                    type='' />
            </Playground>
        </div>
    );
};
