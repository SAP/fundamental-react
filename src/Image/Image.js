import PropTypes from 'prop-types';
import React from 'react';

export const Image = ({ size, type, photo, className, ...props }) => {
    return (
        <span
            className={`${'fd-image--' + size}${type ? ' fd-image--' + type : ''}${className ? ' ' + className : ''}`}
            style={{ backgroundImage: 'url(' + photo + ')' }} {...props} />
    );
};

Image.propTypes = {
    photo: PropTypes.string.isRequired,
    size: PropTypes.oneOf(['s', 'm', 'l']).isRequired,
    className: PropTypes.string,
    type: PropTypes.oneOf(['', 'circle'])
};
