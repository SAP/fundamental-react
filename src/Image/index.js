export { default as Image } from './Image.js';
