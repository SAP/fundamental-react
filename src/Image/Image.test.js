import { Image } from './Image';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

describe('<Image />', () => {
    const image = (
        <Image
            className='blue'
            photo='https://placeimg.com/400/400/nature'
            size='s' />
    );
    const circleImage = (
        <Image photo='https://placeimg.com/400/400/nature' size='m'
            type='circle' />
    );
    test('create image', () => {
        let component = renderer.create(image);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        component = renderer.create(circleImage);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the Image component', () => {
            const element = mount(
                <Image
                    data-sample='Sample'
                    photo='https://placeimg.com/400/400/nature'
                    size='s' />
            );

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
