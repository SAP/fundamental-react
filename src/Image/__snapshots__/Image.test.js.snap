// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> create image 1`] = `
<span
  className="fd-image--s blue"
  style={
    Object {
      "backgroundImage": "url(https://placeimg.com/400/400/nature)",
    }
  }
/>
`;

exports[`<Image /> create image 2`] = `
<span
  className="fd-image--m fd-image--circle"
  style={
    Object {
      "backgroundImage": "url(https://placeimg.com/400/400/nature)",
    }
  }
/>
`;
